### ISSUE TITLE
Keeping a forceNetwork centered in a Shiny app?

### ISSUE THREAD

**Issue by VMJim** [christophergandrud/networkD3#176]
When displaying a forceNetwork in a Shiny app in which opacity is a reactive value, the graph rerenders but in such a way that the network moves out of view towards the upper left corner. 

Below are two examples:

 * ```shiny::runGitHub('christophergandrud/networkD3-shiny-example')```

 * http://stackoverflow.com/questions/42764760/how-to-keep-a-forcenetwork-networkd3-centered-in-a-shiny-app/42767469#42767469

It would be nice, when the nodes do not move when the opacity changes.

**Comment by cjyetman** [comment]
This is a combination of two things that seem to have been introduced by the switch to D3v4...

1. In the D3v4 version, the start position of nodes is in the top left corner, whereas in the D3v3 version, that start positions appears to be in the center (or possibly evenly spread across the plot?). This is likely related to D3v4's effort to make d3.force() have deterministic output by having set start positions.
2. In the D3v3 version, the force timer seems to get reset every time shiny sets a new value and rerenders. In the D3v4 version, the force timer continues and eventually runs out even while shiny is passing in new values. We may have to manually restart and 'reheat' the plot on render to get around this. Not sure why D3v4 is different in this case.

**Comment by cjyetman** [comment]
setting d3.forceX() and d3.forceY() to half the width and half the height when the simulation is initiated will probably resolve issue 1 above

**Comment by cjyetman** [comment]
@VMJim I think I fixed both of these issues. Please try installing my dev branch and testing to see if you get an improved result using...
```r
devtools::install_github('cjyetman/networkD3', ref = 'center_forceNetwork')
```
If this works, I'll make a PR for @christophergandrud to review.

**Comment by VMJim** [comment]
Your fix works (thanks!). And I would say the new behavior is preferred to the throw-from-upper-left-corner.

p.s. Complete integration with Shiny's reactivity would be even more awesome... :D (dreaming... of the ShinyNetwokD3 package)

**Comment by cjyetman** [comment]
@christophergandrud looks like another branch is potentially ready for merge... it would resolve #176 (this one) and probably #109 and #174. Let me know if you think I should prep the PR

**Comment by cjyetman** [comment]
@VMJim I'm not sure exactly what you mean by "complete integration with Shiny's reactivity", so feel free to elaborate in a new issue/feature request.

**Comment by christophergandrud** [comment]
Looks great @cjyetman. Please do merge

**Comment by cjyetman** [comment]
this issue is resolved by #179... closing
